tox.ini

```diff
@@ -21,80 +24,121 @@ extras =
     pytest --doctest-modules src/RestrictedPython/compile.py {posargs}
 
 setenv =
   COVERAGE_FILE=.coverage.{envname}
 
 deps =
+    -cconstraints.txt
     pytest-cov
     pytest-remove-stale-bytecode
     pytest-html
 
 [testenv:py27-datetime]
 basepython = python2.7
 deps =
+    -cconstraints.txt
     {[testenv]deps}
     DateTime
 
 [testenv:py36-datetime]
 basepython = python3.6
 deps =
+    -cconstraints.txt
     {[testenv]deps}
     DateTime
 
 [testenv:py27-rp3]
 basepython = python2.7
 commands =
-    coverage run {envbindir}/zope-testrunner --path=src/RestrictedPython --all -v {posargs}
+    coverage run {envbindir}/zope-testrunner --path=src/RestrictedPython --all {posargs}
 deps =
+    -cconstraints.txt
     .[test]
     zope.testrunner
     coverage
 
 [testenv:coverage]
 basepython = python2.7
-deps = coverage
+skip_install = true
+
+deps =
+    -cconstraints.txt
+    coverage
+
 setenv =
   COVERAGE_FILE=.coverage
 
-skip_install = true
 commands =
     coverage erase
     coverage combine
     coverage html
     coverage xml
     coverage report
 
-[testenv:isort]
-basepython = python2.7
-deps = isort
-commands =
-    isort --check-only --recursive {toxinidir}/src {toxinidir}/tests {posargs}
-
 [testenv:isort-apply]
 basepython = python2.7
-deps = isort
+skip_install = true
+deps =
+    -cconstraints.txt
+    isort
+
 commands =
     isort --apply --recursive {toxinidir}/src {toxinidir}/tests {posargs}
 
-[testenv:flake8]
-basepython = python2.7
+[lint]
+skip_install = true
+
 deps =
+    -cconstraints.txt
     flake8
+    # helper to generate HTML reports:
     flake8-html
-    flake8-debugger
-    flake8-todo
+    # Useful flake8 plugins that are Python and Plone specific:
+    isort
 
 commands =
-    - flake8 --format=html --htmldir={toxinidir}/reports/flake8 --doctests src tests setup.py {posargs}
-    flake8 --doctests src tests setup.py {posargs}
+    mkdir -p {toxinidir}/reports/flake8
+    isort --check-only --recursive {toxinidir}/src {toxinidir}/tests setup.py
+    - flake8 --format=html --htmldir={toxinidir}/reports/flake8 --doctests src tests setup.py
+    flake8 src tests setup.py --doctests
+
+whitelist_externals =
+    mkdir
+
+[testenv:lint-py2]
+basepython = python2
+skip_install = true
+deps = {[lint]deps}
+commands = {[lint]commands}
+whitelist_externals = {[lint]whitelist_externals}
+
+[testenv:lint-py3]
+basepython = python3
+skip_install = true
+deps = {[lint]deps}
+commands = {[lint]commands}
+whitelist_externals = {[lint]whitelist_externals}
 
 [testenv:docs]
 basepython = python2.7
 commands =
     sphinx-build -b html -d build/docs/doctrees docs build/docs/html
     sphinx-build -b doctest docs build/docs/doctrees
 deps =
-    .[docs]
+    -cconstraints.txt
+    Sphinx
+
+[testenv:release]
+skip_install = true
+basepython = python2.7
+
+deps =
+    -cconstraints.txt
+    zest.releaser[recommended]
+
+commands =
+    python -V
+    fullrelease --no-input -v
```

Comment:
I do not like to have this one here, especially that `fullrelease` runs without asking questions, this is a no-go for a security package. The developer releasing this package should have the right tools to do it in their day-to-day tool chain.

Reply:
It is an optional helper that makes sense for small bugfix releases. If you do not like it, you do not need to use it. 

And it would also only work if every test of zest.releaser passes and you have publish right to pypi.